src/lib/user/actions/get-my-account.ts

```diff
@@ -0,0 +1,48 @@
+import { UserActor } from "../user-actor";
+import { dbQuery } from "@/lib/data/db-utils";
+import { MyAccount } from "../user-models";
+
+export async function getMyAccount(
+  actor: UserActor,
+): Promise<MyAccount | null> {
+  const { userId: user_id, dbPool, userMapper } = actor.getParams();
+
+  const sql = `
+  SELECT
+    user_id as "userId",
+    user_creation_time as "userCreationTime",
+    user_hashed_email as "userHashedEmail",
+    user_encrypted_pii as "userEncryptedPii",
+    user_residency as "userResidency"
+  FROM 
+    users
+  WHERE 
+    user_id = $1
+  `;
+  const res = await dbQuery(dbPool, sql, [user_id]);
+  if (res.rows.length === 0) {
+    return null;
+  }
+  const {
+    userId,
+    userCreationTime,
+    userHashedEmail,
+    userEncryptedPii,
+    userResidency,
+  } = res.rows[0];
```

Comment:
We can reduce the repetition by using the "rest pattern". In this case, we only need `userHashedEmail` and `userEncryptedPii` for some post processing, so we can destructure it like this.

```ts
const {userHashedEmail, userEncryptedPii, ...otherFields} = res.rows[0];
```

So after we've gotten the PII, we can

```ts
return {userName, userEmail, userPhoneNumber, ...otherFields};
```

This way when we add new fields to MyAccount that isn't PII, we only need to update the SQL.

Reply:
Done! b07f27143d693cce888d59d16c08072d117813f4